src/main/java/de/tum/in/www1/artemis/web/rest/QuizExerciseResource.java

```diff
@@ -277,16 +298,110 @@ public ResponseEntity<QuizExercise> recalculateStatistics(@PathVariable Long qui
+     * @return the ResponseEntity with status 200 (OK) and with body the quizBatch that was joined
+     */
+    @PostMapping("/quiz-exercises/{quizExerciseId}/join")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<QuizBatch> joinBatch(@PathVariable Long quizExerciseId, @RequestBody QuizBatchJoinDTO joinRequest) {
+        log.debug("REST request to join Batch : {}", quizExerciseId);
+        QuizExercise quizExercise = quizExerciseRepository.findByIdElseThrow(quizExerciseId);
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAllowedToSeeExercise(quizExercise, user)) {
+            throw new AccessForbiddenException();
+        }
+        if (quizScheduleService.getQuizBatchForStudent(quizExercise, user).isPresent()) {
+            return ResponseEntity.badRequest()
+                    .headers(HeaderUtil.createFailureAlert(applicationName, true, "quizExercise", "quizBatchPending", "Previous submission for this quiz is still pending."))
+                    .build();
+        }
+
+        var submissions = submissionRepository.countByExerciseIdAndStudentId(quizExerciseId, user.getId());
+        if (quizExercise.getAllowedNumberOfAttempts() != null && submissions >= quizExercise.getAllowedNumberOfAttempts()) {
+            return ResponseEntity.badRequest()
+                    .headers(HeaderUtil.createFailureAlert(applicationName, true, "quizExercise", "quizAttemptsExceeded", "Maximum number of attempts reached.")).build();
+        }
+
+        var quizBatch = quizBatchService.joinBatch(quizExercise, user, joinRequest.password, joinRequest.batchId);
+
+        if (quizBatch.isEmpty()) {
+            return ResponseEntity.notFound().headers(HeaderUtil.createFailureAlert(applicationName, true, "quizExercise", "quizBatchNotFound", "Quiz batch not found or invalid."))
+                    .build();
+        }
+
+        return ResponseEntity.ok(quizBatch.get());
+    }
+
+    /**
+     * POST /quiz-exercises/:quizExerciseId/add-batch : add a new batch to the exercise for batched running
+     *
+     * @param quizExerciseId the id of the quizExercise to add the batch to
+     * @return the ResponseEntity with status 200 (OK) and with body the new batch
+     */
+    @PutMapping("/quiz-exercises/{quizExerciseId}/add-batch")
+    @PreAuthorize("hasRole('TA')")
+    public ResponseEntity<QuizBatch> addBatch(@PathVariable Long quizExerciseId) {
+        log.debug("REST request to add batch : {}", quizExerciseId);
+        QuizExercise quizExercise = quizExerciseRepository.findByIdWithQuestionsElseThrow(quizExerciseId);
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(quizExercise, user)) {
+            throw new AccessForbiddenException();
+        }
+
+        // TODO: QQQ configuration to limit the number of batches a tutor can create
```

Comment:
please describe such TODOs in more detail if you cannot resolve them now.
What does QQQ stand for?

Reply:
nothing in particular other than Q for quiz; those are the the locations where i want to fix extend things for the followup / cleanup pass st that i can find those TODOs in particular again.

Reply:
I would recommend using e.g. @bottbenj inside the TODO so that it is also clear to other developers that you are still working on it

Reply:
i have changed it to TODO: quiz cleanup but ill keep that in mind